### PR TITLE
Script for root transition.

### PR DESCRIPTION
**Pull request by myidpt**

### CODE REVIEW

File: bin/root-transition.sh [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  NS=`kubectl get ns | grep -v "STATUS" | grep -v "kube-system" | grep -v "kube-public" | awk '{print $1}'`
+
+  for n in $NS
+  do
+    echo "Checking namespace: $n"
+    verify_namespace $n
+  done
+
+  if [ -z $NOT_UPDATED ]; then
+    echo "------All Istio keys and certificates are updated in secret!"
```
Comment (mandarjog) [inline bin/root-transition.sh]: If you run verify twice, the message says that `certificate are updated` and that can feel like you just changed something.

It should say "certificates are up to date, or state the expiration date in the future?

File: bin/root-transition.sh [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script extends the lifetime of the self-signed Citadel root certificate in the current cluster.
+# It offers the following options:
+# - check: Check the current root certificate lifetime.
+# - transition: Extend the lifetime of the current root certificate.
+# - verify: Check the new workload certificates are generated.
+
```
Comment (mandarjog) [inline bin/root-transition.sh]: mention that this script required kubectl, openssl and `bc`.

I did not have `bc` in mycluster so the script failed.

File: bin/root-transition.sh (hunk 1 above)
Comment (mandarjog) [reply bin/root-transition.sh]: I ran verify before running anything and it said everything was fine.
Sounds like there is very little difference between check and verify.

Comment (myidpt) [reply bin/root-transition.sh]: "Verify" verifies the root certs propagated to the workload secrets are the same as the root cert used in Citadel.
"Check" checks the current root cert lifetime.
I can rename the command if necessary.

Comment (myidpt) [reply bin/root-transition.sh]: Agree to improve the message.

File: bin/root-transition.sh (hunk 2 above)
Comment (myidpt) [reply bin/root-transition.sh]: Done. Thanks.

**Comment by costinm** [comment]
Also would be good to avoid bc. Expr ? Bash expressions ?

**Pull request merged**